src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -0,0 +1,584 @@
+                            // case "calendar_dates":
+                            case "calendar":
+                                // If any service_id in the active feed matches with the future feed, it should be
+                                // modified and all associated trip records must also be changed with the modified
+                                // service_id.
+                                // TODO How can we check that calendar_dates entries are duplicates? I think we would
+                                //  need to consider the service_id:exception_type:date as the unique key and include any
+                                //  all entries as long as they are unique on this key.
+                                for (NewGTFSError error : idErrors) {
+                                    if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) {
+                                        String key = String.join(":", table.name, idScope, val);
+                                        // Modify service_id and ensure that referencing trips have service_id updated.
+                                        val = String.join(":", idScope, val);
+                                        mergeFeedsResult.remappedIds.put(key, val);
+                                    }
+                                }
+                                // If a service_id from the active calendar has both the start_date and end_date in the
+                                // future, the service will be excluded from the merged file. Records in trips,
+                                // calendar_dates, and calendar_attributes referencing this service_id shall also be
+                                // removed/ignored. Stop_time records for the ignored trips shall also be removed.
+                                if (feedIndex > 0) {
+                                    int startDateIndex = getFieldIndex(fieldsFoundInZip, "start_date");
+                                    LocalDate startDate = LocalDate.parse(csvReader.get(startDateIndex), GTFS_DATE_FORMATTER);
+                                    if (startDate.isAfter(LocalDate.now())) {
+                                        LOG.warn("Skipping calendar entry {} because it operates in the future.", keyValue);
+                                        String key = String.join(":", table.name, idScope, keyValue);
+                                        mergeFeedsResult.skippedIds.add(key);
+                                        skipRecord = true;
+                                        continue;
+                                    }
+                                    // If a service_id from the active calendar has only the end_date in the future, the
+                                    // end_date shall be set to one day prior to the earliest start_date in future
+                                    // dataset before appending the calendar record to the merged file.
```

Comment:
Again, I don't understand why this is being done.

Reply:
I think this might be ok after I realized what is happening down at https://github.com/conveyal/datatools-server/pull/186/files#r272345429. If so, mark this as resolved and let's discuss at that other comment.